packages/ag-charts-enterprise/src/features/sync/chartSync.ts

```diff
@@ -136,12 +135,13 @@ export class ChartSync extends BaseProperties implements _ModuleSupport.ModuleIn
                 .map(this.findMatchingNodes(axis, mainDirection, valueIsDate, eventValue))
                 .filter(isDefined);
 
-            if (this.domainMode === 'key' && hasKeyBasedDomains) {
-                const secondaryKey = `${secondaryDirection}Key` as const;
-                const secondaryValue = _ModuleSupport.isObjectWithProperty(event.currentHighlight, secondaryKey)
-                    ? event.currentHighlight?.[secondaryKey]
-                    : undefined;
-                // Narrow matches by matching the secondary direction key.
+            const secondaryKey = `${secondaryDirection}Key` as const;
+            const secondaryValue = _ModuleSupport.isObjectWithProperty(event.currentHighlight, secondaryKey)
+                ? event.currentHighlight?.[secondaryKey]
+                : undefined;
```

Comment:
nit: shouldn't it be inside the if block? it seems to be relevant only when `useSecondaryDirectionKey === true`

Reply:
Fair, updated.